### ISSUE TITLE
DHCP ignores device checksum capabilities

### ISSUE THREAD

**Issue by Spadi0** [smoltcp-rs/smoltcp#371]
`Dhcpv4Client` ignores the device checksum capabiltities, and always uses the default value for `ChecksumCapabilities`. The code responsible can be seen here:
https://github.com/smoltcp-rs/smoltcp/blob/bdfa44270e9c59b3095b555cdf14601f7dc27794/src/dhcp/clientv4.rs#L133

As far as I can tell, this would basically be a one-line fix:
```rust
let checksum_caps = iface.device().capabilities().checksum;
```

This code breaks DHCP for any device that doesn't use the default checksum capabilities.

**Comment by birkenfeld** [comment]
"breaks" is a bit dramatic - didn't it just mean that smoltcp always calculates the checksums itself?

**Comment by Spadi0** [comment]
@birkenfeld 
It depends on how the driver handles checksums - if the driver expects the checksum field to be empty, and it is not, then the calculated checksum becomes invalid. I have tested this on a STM32F429NIHx, and IIRC, this is the case on a lot of embedded MCUs with an ethernet peripheral.

As I would imagine that one of the more popular use cases for smoltcp is embedded systems, that is why I used "breaks"; although, as you said, maybe that word is a bit dramatic.

Either way though, I can't see a reason to not change this code. As there is a comment with a question mark, I presume this was boilerplate used when writing the DHCP module that was intended to be changed before it was merged, but it just slipped through the cracks.
